### PR TITLE
[#1354] Revert experimental GroovyTempate.compile code to fix template issue

### LINKED ISSUE #1354: The temporary class of the template is output directly under the application dir.
### Are you looking for help?

This is an issue tracker, used to manage and track the development of Play. It is not a support system and so it is not a place to ask questions or get help. If you're not sure if you have found a bug, or if you have a feature request, the best place to start is with either the [Discuss Play Forum](https://discuss.playframework.com) or [Stack Overflow](http://stackoverflow.com/questions/ask?tags=playframework).

### Play Version (1.5.x / etc)
1.6.0

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)
windows/Centos

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)
OpenJDK 11

### 

The temporary class of the groovy template will be output directly under the application dir.
I want them to be output to the tmp dir.

```
@@ -118,6 +118,7 @@ public class GroovyTemplate extends BaseTemplate {
     protected CompilerConfiguration setUpCompilerConfiguration() {
         CompilerConfiguration compilerConfiguration = new CompilerConfiguration();
         compilerConfiguration.setSourceEncoding("utf-8"); // ouf
+        compilerConfiguration.setTargetDirectory("tmp");
         return compilerConfiguration;
     }
 ```

### PR DESCRIPTION
**Pull request by Maximillian13**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [ ] Have you updated the documentation?
* [ ] Have you added tests for any changed functionality?

# Helpful things
Issue Discussion: https://github.com/playframework/play1/issues/1354
Breaking commit: https://github.com/playframework/play1/pull/1338/commits/8e2e363c7d894eb7e99c1b28cf61e01aedcd8e28

## Fixes
Fixes #1354 

## Purpose

This pull request:
- Revert "experimental" (code with a comment discussing not knowing the full impact) GroovyTempate.compile code to fix template files saving to projects root

## Background Context
This was discussed in the linked thread. Reverting back to the old behavior seems reasonable as well as we know it was working properly and was changed to get rid of reflection, not to fix a bug.

We may want to add an improvement ticket to look back at this area and re-write the reflection code because reflection is usually a bad idea.  

## References
https://github.com/playframework/play1/issues/1354
https://github.com/playframework/play1/pull/1338/commits/8e2e363c7d894eb7e99c1b28cf61e01aedcd8e28

### CODE REVIEW

File: framework/src/play/templates/GroovyTemplate.java [hunk 1]
```diff
@@ -139,34 +139,18 @@ public void compile() {
...
-				// The following approach to adding the phase operation replaces the original
-				// reflection based approach commented out lower down. This appears to be the
-				// canonical approach and possibly has only been made available in the v3.x
-				// stream but it differs in two ways from the reflection based approach and it's
-				// not clear if and what the impact is:
-				// 1. It does NOT guarantee an empty list of OUTPUT phases operations to begin with.
-				// 2. The new phase operation is added to the start and not the end.
-                // See https://github.com/apache/groovy/blob/GROOVY_3_0_6/src/main/java/org/codehaus/groovy/control/CompilationUnit.java#L349
-                compilationUnit.addPhaseOperation(new IGroovyClassOperation() {
+
```
Comment (tazmaniax) [inline framework/src/play/templates/GroovyTemplate.java:142]: Add this comment

```
// Play needs to handle writing the generated Groovy class to the file system but the Groovy
// compilation unit by default adds it's own output phase operation to do this that cannot
// be replaced using the available public methods. Until Groovy provides this capability
// it's necessary to access the compilation unit directly using reflection to replace the
// default output operation with the Play Groovy class handler.
```

**Review by tazmaniax — COMMENTED** [review]
It is correct to revert this change and I would just add the following comment to avoid this change happening again in the future.

**Comment by tazmaniax** [comment]
Just because it's taking a while to get this released I've applied this change to my JDK 17 compatible Play1 [pull request](https://github.com/tazmaniax/play1/pull/9, which also now replaces Log4j with reload4j.

**Review by xael-fry — APPROVED** [review]

**Comment by willhains** [comment]
🎉 Thank you! ❤️

**Pull request merged**